Rewrite Game_mode/osu! (#13645)
ppy/osu-wiki: wiki/Game_mode/osu!/en.md

# osu! (game mode)

*For other uses, see [osu! (disambiguation)](/wiki/Disambiguation/osu!).*

> osu! is the first game mode introduced to the game. In it, the player has the ability to aim for and click circles using their *cursor* in tune with the rhythm.

Author: i can see your point so ill use this, but instead of using `(formerly known as *osu!standard*)` I'll use `, formerly known as *osu!standard*,` since it was done this way in the catch article and we might as well keep this consistent for something as minor as this.

When the [auto pilot](/wiki/Gameplay/Game_modifier/Autopilot) mod is being used, the in-game cursor will automatically aim for all objects, only requiring the user to click to the rhythm. The speed the spinners will spin at is the same as the one in the [spun out](/wiki/Gameplay/Game_modifier/Spun_Out) mod.

If the player wants to draw while playing osu!, they can do using **smoke**. When the player activates the respective key for smoke, the cursor will leave behind a white trail that will last for about 8 seconds after the keys deactivation.

<!-- TODO: Create a article dedicated to smokes. -->

<!-- This part describing smokes could be expanded into its separate section instead of being here just to describe what the smoke key is for. Leaving this here for later usage in such section: ![osu! smoke effect](/wiki/shared/osu_smoke.jpg "Smoke Usage") -->

## Hit objects

osu! has 3 main [hit objects](/wiki/gameplay/Hit_object) the player interacts with:

- Hit circles
- Sliders
- Spinners

### Hit circles

> If the judgement is everything but a `MISS`, the combo will be incremented by 1.

Author: This is what I get for having a broken keyboard that doubles my ctrl+v 😞

# osu! (game mode)

*For other uses, see [osu! (disambiguation)](/wiki/Disambiguation/osu!).*

osu! is the first game mode introduced to the game. In it, the player has the ability to aim for and click circles using their *cursor* in tune with the rhythm.

## Song selection

To access the osu! game mode press `Ctrl` + `1` at the same time, use the the `Mode` button and select `osu! in osu!(stable), or click the ![](/wiki/shared/mode/osu.png) icon on the top of the screen for osu!(lazer).

To download more beatmaps for osu!, head to the [beatmap listing](https://osu.ppy.sh/beatmapsets) on the osu! website, or while in osu!(lazer) press `Ctrl` + `B` or click the `beatmap listing` button.

![osu!(lazer) beatmap listing button](/wiki/shared/lazer_beatmap_listing.png)

## Player control

In osu!, the user is supposed to click circles, guide sliders and spin spinners by using their cursor or touch screen. By default the keys to control the cursors actions are:

- The `left mouse button` and `right mouse button` or `Z` and `X` to click using cursor.

> - The `C` button to draw with smoke.

Author: there doesn't seem to be one, this is something I might look into after this

For touch screen devices, simply tap the hit objects or aim with one hand and tap to the rhythm in the area outside the playfield with the other.

> If [relax](/wiki/Gameplay/Game_modifier/Relax) is being used, the in-game cursor will automatically click the hit objects when aimed to the rhythm. Spinners still have to be completed manually.

Author: relax and autopilot are mentioned in the previous version of the article, especially relax that is mentioned across all gamemodes (except mania but that doesn't count). 

relax and autopilot remove core game mechanics, unlike mods like HD, HT, or even ST. Most mods in the game *alter* how the rulesets mechanics work, instead of straight up removing them, thats why I'd personally keep them in all gamemode articles.

(also relax is also mentioned in the new catch article therefore this could be done for consistency across articles too.)

After hitting a hit circle, a number appears indicating the [judgement](/wiki/Gameplay/Judgement/osu!) received for how accurate the timing of the hit was.
If the judgement is everything but a `MISS`, the combo will be incremented by 1.

### Sliders

![Sliders](/wiki/shared/osu_slider.jpg "osu! sliders")

**Sliders** build on hit circles mechanics. Firstly, the player has to click the circle at the beginning of the slider, called the [slider head](/wiki/Gameplay/Hit_object/Slider/Sliderhead), when the approach circle touches it. After hitting the slider head, a ball will begin to move across the *slider path*. The outer circle, called the follow circle, will appear when holding onto the slider's ball, but will disappear when the cursor is outside the follow circle or when the ball is released.

Sometimes, the ball may reverse its direction after hitting a *reverse arrow* at the end of the current slider path. The ball may can hit a reverse arrow more than once, making it go back from the start to the end once more.

### Spinners

![Spinner](/wiki/shared/osu_spinner.jpg "osu! spinners")

> **Spinners** are the simplest object type in osu!. To clear a spinner, the user has to activate the in-game cursor and spin it around the center as fast as they can. If the spinner was cleared early, you can continue spinning to acquire bonus score and regain more health. Spinners don't have hit windows, but will break combo and grant a `MISS` judgement when not cleared properly.

Author: > (by default `X` or `Z`)

considering the default keybinds are listed above `Hit objects` I don't think there is a reason to mention them again here

> **Spinners** are the simplest object type in osu!. To clear a spinner, the user has to activate the in-game cursor and spin it around the center as fast as they can. If the spinner was cleared early, you can continue spinning to acquire bonus score and regain more health. Spinners don't have hit windows, but will break combo and grant a `MISS` judgement when not cleared properly.

Author: I missed your edit mb I'll get it next time TvT.

> Below the spinner you can see your RPM, which shows the current spin speed thats measured in spins per minute.

Author: will try my best to remeber this

On certain skins, the outer white circle will turn red to notify that time to clear the spinner is almost out. Skins using [skin version](/wiki/Skinning/skin.ini#versions) 1.0, will have a meter/gauge to indicate how close the spinner is to being completed.

## Difficulty settings

osu! has 4 difficulty settings with each affecting the gameplay differently:

- [Circle size](/wiki/Beatmap/Circle_size)
- [HP drain](/wiki/Beatmap/HP_drain_rate)
- [Approach rate](/wiki/Beatmap/Approach_rate)
- [Overall difficulty](/wiki/Beatmap/Overall_difficulty)

> Each setting is often referred to as CS, HP, AR, and OD respectively.

Author: I in fact did something with this

> Each setting is often referred to as CS, HP, AR, and OD respectively.

Reviewer: This is fine for me. Sorry for answering so late. I did not realise you have added a comment.

*Refer to [Play Styles page under osu!](/wiki/Gameplay/Play_style).*

## Scoring

[Score in osu!](/wiki/Gameplay/Score/ScoreV1/osu!) is a weighted sum of multiple components of gameplay. It depends on the following:

- [Judgement](/wiki/Gameplay/Judgement/osu!) determines a hit object's base scoring value (300, 100, 50, or 0 in case of a miss). For hit circles and slider heads, well-timed key presses are valued more, both in terms of score and accuracy. Getting a higher judgement also provides more [health](/wiki/Gameplay/Health) boost.
- [Accuracy](/wiki/Gameplay/Accuracy#osu!) is dependant on judgments and shows how precise hits are. Late or early key presses, as well as misses, decrease overall accuracy.
- [Combo](/wiki/Gameplay/Combo_(score_multiplier)) is a score multiplier: clearing a hit object contributes more to the total score when combo is high and vice versa. Combo may be [broken](/wiki/Gameplay/Judgement/Combobreak) by a miss or a [slider break](/wiki/Gameplay/Judgement/Slider_break).

In osu!(stable), slider heads also didn't have hit timings and the sliders judgement was decided when the slider ended, with it being based on how the player guided the slider ball.

> When combo is maintained, the total score grows exponentially. Objects closer to the end of the map are worth more points than the ones in the beginning, which means that a player will lose more potential score on them in case of mistimed hits. As a result, it's possible for a score with lower accuracy to have a higher amount of points and beat a score with higher accuracy.

Author: I always forget about not using those lol

# osu! (game mode)

*For other uses, see [osu! (disambiguation)](/wiki/Disambiguation/osu!).*

> **osu!**, formerly known as *osu!standard*, is the default game mode in which players are required to aim for and click circles using their *cursor* in tune with the rhythm.

Reviewer 2: *formerly known as what??**

It was never ever officially known as this, so I'm not sure what this is intending to describe.

![osu!(lazer) beatmap listing button](/wiki/shared/lazer_beatmap_listing.png)

## Player control

In osu!, the user is supposed to click circles, guide sliders and spin spinners by using their cursor or touch screen. By default the keys to control the cursor's actions are:

- The `left mouse button` and `right mouse button` or `Z` and `X` to click using cursor.
- The `C` button to draw with smoke.

For touch screen devices, simply tap the hit objects or aim with one hand and tap to the rhythm in the area outside the playfield with the other.

If [relax](/wiki/Gameplay/Game_modifier/Relax) is being used, the in-game cursor will automatically click the hit objects when aimed to the rhythm. Spinners still have to be completed manually.

When the [auto pilot](/wiki/Gameplay/Game_modifier/Autopilot) mod is being used, the in-game cursor will automatically aim for all objects, only requiring the user to click to the rhythm. The speed the spinners will spin at is the same as the one in the [spun out](/wiki/Gameplay/Game_modifier/Spun_Out) mod.

> If the player desires to draw while playing osu!, they can do so using **smoke**. When the player activates the respective key for smoke, the cursor will leave behind a white trail that will last for about 8 seconds after the keys deactivation.

Reviewer: ```suggestion
If the player desires to draw while playing osu!, they can do so using **smoke**. When the player activates the respective key for smoke, the cursor will leave behind a white trail that will last for about 8 seconds after the key's deactivation.
```

Below the spinner is a display showing the current spin speed in revolutions per minute (RPM).

On certain skins, the outer white circle will turn red to notify that time to clear the spinner is almost out. Skins using [skin version](/wiki/Skinning/skin.ini#versions) 1.0, will have a meter/gauge to indicate how close the spinner is to being completed.

## Difficulty settings

osu! has 4 difficulty settings with each affecting the gameplay differently:

- [Circle size](/wiki/Beatmap/Circle_size) (CS): Determines how big the hit circles are.
- [HP drain](/wiki/Beatmap/HP_drain_rate) (HP): Defines how much health is gained and lost while playing.
- [Approach rate](/wiki/Beatmap/Approach_rate) (AR): Regulates the speed hit objects appear in.
- [Overall difficulty](/wiki/Beatmap/Overall_difficulty) (OD): Controls the strictness of timings.

> Each setting is often referred to as CS, HP, AR, and OD respectively. Click on the linked articles above for more detailed information.

Reviewer: Since the attributes are described and linked above, I don't think we still need this line. (Second phrase is kinda self-explanatory now).

Author: I'll keep em

- On the *DS*, completing spinners is a good way to scratch the touchscreen (or screen protector), especially on the higher difficulties. In osu!, spinners have been toned down as to not be such a nuisance to complete. In osu!(stable), the maximum number of spins per minute achievable is 477 which is also the rate at which the [Auto](/wiki/Gameplay/Game_modifier/Auto) game modifier spins while the [Spun Out](/wiki/Gameplay/Game_modifier/Spun_Out) game modifier spins slower at 287 spins per minute. In osu!(lazer), spinners have set maximum for bonus score, making spinning after achieving the max score unnecessary.
- Changing spin direction in the middle of a spin will cause the spinner to slow down (spins per minute will go down to 0 to reset the spin orientation), then spin in the correct direction. The spin progress won't progress during the reorientation, and will continue to increase on reverse spin.
- Not spinning around the centre core of the spinner will **not** be considered as a valid spin.

> - Smoke last longer than default in-game cursor trail. Apply smoke effects sparsely to prevent old smoke effects disappearing too quickly.

Reviewer: I was talking about the smoke in line 120 😅

My question was what "sparsely apply smoke" means.

> - Smoke last longer than default in-game cursor trail. Apply smoke effects sparsely to prevent old smoke effects disappearing too quickly.

Author: it seems the same, ill drop this

- The osu! mascot is [pippi](/wiki/Mascots#pippi).

![osu! smoke effect](/wiki/shared/osu_smoke.jpg "Smoke Usage")

### History

> 

Author: > Is there a reason why the song names are mentioned twice, once with a link and once in a code block?

I think I just forgor to drop these, oh well

- In osu!(stable), the maximum number of spins per minute achievable is 477, which is also the rate at which the [auto](/wiki/Gameplay/Game_modifier/Auto) game modifier spins. In osu!(lazer), spinners have a set maximum for bonus score, making spinning after achieving the max score unnecessary. The [spun out](/wiki/Gameplay/Game_modifier/Spun_Out) game modifier spins slower at 287 spins per minute in both versions of the game. The reason for these limits being set is because on the original DS games, completing spinners is a good way to scratch the touchscreen (or its screen protector), especially on higher difficulties. These limits protect the player from potentially damaging their devices.
- Changing the spin direction in the middle of a spin will cause the spinner to slow down (spins per minute will go down to 0 to reset the spin orientation), then start spinning in the correct direction. The spin progress will not progress during the reorientation.

> - Not spinning around the core of the spinner will **not** be considered as a valid spin.

Reviewer: ```suggestion
- Clearing a spinner is **only** possible by spinning around its core.
```
or

```suggestion
- A spin is only considered valid if spinning around the core.
```

Might read better

Author: Decided on `Clearing a spinner is **only** possible by spinning around its core, spinning beside it is not valid and will not grant score.`, I think it somewhat clarifies it
